EXILED/Exiled.Events/Patches/Generic/Scp079Recontain.cs

```diff
@@ -0,0 +1,54 @@
+// -----------------------------------------------------------------------
+// <copyright file="Scp079Recontain.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Generic.Scp079API
+{
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using API.Features.Pools;
+
+    using HarmonyLib;
+
+    using PlayerRoles.PlayableScps.Scp079;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    /// Patches <see cref="Scp079Recontainer.OnServerRoleChanged(ReferenceHub, PlayerRoles.RoleTypeId, PlayerRoles.RoleChangeReason)"/>.
+    /// Adds the <see cref="Exiled.Events.Config.ShouldScp079RecontainedWhenNoScps" /> support.
+    /// </summary>
+    [HarmonyPatch(typeof(Scp079Recontainer), nameof(Scp079Recontainer.OnServerRoleChanged))]
+    internal class Scp079Recontain
+    {
+        private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions, ILGenerator generator)
+        {
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Pool.Get(instructions);
+
+            Label ret = generator.DefineLabel();
+
+            newInstructions.InsertRange(
+                0,
+                new[]
+                {
+                    // if (!Events.Instance.Config.ShouldScp079RecontainedWhenNoScps)
+                    //     return;
+                    new CodeInstruction(OpCodes.Call, PropertyGetter(typeof(Exiled.Events.Events), nameof(Exiled.Events.Events.Instance))),
+                    new(OpCodes.Callvirt, PropertyGetter(typeof(Exiled.Events.Events), nameof(Exiled.Events.Events.Config))),
+                    new(OpCodes.Callvirt, PropertyGetter(typeof(Exiled.Events.Config), nameof(Exiled.Events.Config.ShouldScp079RecontainedWhenNoScps))),
+                    new(OpCodes.Brfalse_S, ret),
```

Comment:
![image](https://github.com/user-attachments/assets/b2717cd6-3ade-45a1-95b5-8fbb941be24d)

Reply:
@IRacle1

Reply:
> @IRacle1

yeah but to make another event, we should change old `Recontain` event logic, bcz `Recontain` is called in both cases(as i remember), and it will be breaking change

Reply:
actually i probably have a solution for it